### PR TITLE
CPDP-507: added select entity type page, form and routes

### PR DESCRIPTION
**Pull request by nubz**

### CODE REVIEW

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/RegistrationController.scala [hunk 1]
```diff
@@ -46,4 +57,44 @@ class RegistrationController @Inject()(
...
+object RegistrationController {
+
+  sealed trait EntityType
+
+  object EntityType {
+    final case object Individual extends EntityType
+    final case object Trust extends EntityType
+  }
+
+  val selectUserTypeForm: Form[EntityType] =
```
[the remark this answers is not in the document]
Comment (nubz) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/RegistrationController.scala]: Done

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/RegistrationController.scala [hunk 2]
```diff
@@ -46,4 +57,44 @@ class RegistrationController @Inject()(
     }
   }
 
+  def selectEntityType():  Action[AnyContent] = authenticatedActionWithSessionData { implicit request =>
+    Ok(selectEntityTypePage(RegistrationController.selectUserTypeForm, routes.RegistrationController.startRegistration()))
```
[the remark this answers is not in the document]
Comment (nubz) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/RegistrationController.scala]: Done

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/RegistrationController.scala [hunk 3]
```diff
@@ -16,34 +16,96 @@
...
+  }
+
+  def selectEntityType():  Action[AnyContent] = authenticatedActionWithSessionData.async { implicit request =>
+    carryOnIfValidUser(request)(_ =>
+      Ok(selectEntityTypePage(RegistrationController.selectEntityTypeForm, routes.RegistrationController.startRegistration()))
+    )
+  }
+
+  def selectEntityTypeSubmit(): Action[AnyContent] =  authenticatedActionWithSessionData { implicit request =>
+    RegistrationController.selectEntityTypeForm.bindFromRequest().fold(
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/RegistrationController.scala]: we could make use of `carryOnIfValid` user here as well if we wanted to be totally anal about it

**Comment by AndyHWChung** [comment]
LGTM

**Pull request merged**